imports/plugins/core/ui/client/components/forms/form.js

```diff
@@ -243,24 +288,33 @@ class Form extends Component {
                 return this.renderField({ fieldName: fullFieldName });
               });
             }
-
             return this.renderField({ fieldName: key });
           }
+          return;
         });
       }
 
       // Render form by only using desired fields from schema
       if (this.props.fields) {
-        return map(this.props.fields, (fieldData, key) => {
+        return Object.keys(this.props.fields).map((key) => { // eslint-disable-line consistent-return
+          const fieldData = this.props.fields[key];
           const fieldSchema = this.schema[key];
+          const tempObj = Object.assign({}, fieldData);
           if (fieldSchema) {
-            return this.renderField({ fieldName: key }, fieldData);
+            // Remove inherited type() as type is supposed to be string.
+            if (typeof tempObj.type === "function") {
+              delete tempObj.type;
+            }
+            const fieldProp = get(fieldsProp, key, undefined);
+            return this.renderField({ fieldName: key }, Object.assign(tempObj, fieldProp));
           }
+          return;
```

Comment:
You do not need to disable `consistent-return` if you change `return` to `return null`.

Also, it is more readable to do early returns rather than large `if` blocks. Something like:

```js
if (!fieldSchema) return null;
// Remove inherited type() as type is supposed to be string.
if (typeof tempObj.type === "function") {
  delete tempObj.type;
}
const fieldProp = get(fieldsProp, key, undefined);
return this.renderField({ fieldName: key }, Object.assign(tempObj, fieldProp));
```

Also, I think it's a bit better to use `{}` as the `fieldProp` default rather than undefined since you are passing it to Object.assign:

```js
const fieldProp = get(fieldsProp, key, {});
```